src/state/staleBalances/index.ts

```diff
@@ -0,0 +1,99 @@
+import { createRainbowStore } from '../internal/createRainbowStore';
+
+const TIME_TO_WATCH = 600000;
+
+interface StaleBalanceInfo {
+  address: string;
+  expirationTime?: number;
+  transactionHash: string;
+}
+
+interface StaleBalances {
+  [key: string]: StaleBalanceInfo;
+}
+interface StaleBalancesByChainId {
+  [key: number]: StaleBalances;
+}
+
+export interface StaleBalancesState {
+  addStaleBalance: ({ address, chainId, info }: { address: string; chainId: number; info: StaleBalanceInfo }) => void;
+  clearExpiredData: (address: string) => void;
+  getStaleBalancesQueryParam: (address: string) => string;
+  staleBalances: Record<string, StaleBalancesByChainId>;
+}
+
+export const staleBalancesStore = createRainbowStore<StaleBalancesState>(
+  (set, get) => ({
+    addStaleBalance: ({ address, chainId, info }: { address: string; chainId: number; info: StaleBalanceInfo }) => {
+      set(state => {
+        const { staleBalances } = state;
+        const staleBalancesForUser = staleBalances[address] || {};
+        const staleBalancesForChain = staleBalancesForUser[chainId] || {};
+        const newStaleBalancesForChain = {
+          ...staleBalancesForChain,
+          [info.address]: {
+            ...info,
+            expirationTime: info.expirationTime || Date.now() + TIME_TO_WATCH,
```

Comment:
do we want to always update to `Date.now() + TIME_TO_WATCH` and ignore the existing `expirationTime` here ?

thinking of a scenario: swap tokenA -> tokenB; and then 9 mins later do tokenC -> tokenB, so we want to keep the "latest" expirationTime